crates/sui-framework/packages/sui-system/sources/validator.move

```diff
@@ -625,73 +634,129 @@ module sui_system::validator {
     }
 
     /// Update name of the validator.
-    public(friend) fun update_name(self: &mut Validator, name: String) {
-        self.metadata.name = name;
+    public(friend) fun update_name(self: &mut Validator, name: vector<u8>) {
+        assert!(
+            vector::length(&name) <= MAX_VALIDATOR_METADATA_LENGTH,
+            EValidatorMetadataExceedingLengthLimit
+        );
+        self.metadata.name = string::from_ascii(ascii::string(name));
     }
 
     /// Update description of the validator.
-    public(friend) fun update_description(self: &mut Validator, description: String) {
-        self.metadata.description = description;
+    public(friend) fun update_description(self: &mut Validator, description: vector<u8>) {
+        assert!(
+            vector::length(&description) <= MAX_VALIDATOR_METADATA_LENGTH,
+            EValidatorMetadataExceedingLengthLimit
+        );
+        self.metadata.description = string::from_ascii(ascii::string(description));
     }
 
     /// Update image url of the validator.
-    public(friend) fun update_image_url(self: &mut Validator, image_url: Url) {
-        self.metadata.image_url = image_url;
+    public(friend) fun update_image_url(self: &mut Validator, image_url: vector<u8>) {
+        assert!(
+            vector::length(&image_url) <= MAX_VALIDATOR_METADATA_LENGTH,
+            EValidatorMetadataExceedingLengthLimit
+        );
+        self.metadata.image_url = url::new_unsafe_from_bytes(image_url);
     }
 
     /// Update project url of the validator.
-    public(friend) fun update_project_url(self: &mut Validator, project_url: Url) {
-        self.metadata.project_url = project_url;
+    public(friend) fun update_project_url(self: &mut Validator, project_url: vector<u8>) {
+        assert!(
+            vector::length(&project_url) <= MAX_VALIDATOR_METADATA_LENGTH,
+            EValidatorMetadataExceedingLengthLimit
+        );
+        self.metadata.project_url = url::new_unsafe_from_bytes(project_url);
```

Comment:
Do we want to make sure these are ascii as well before converting to url?

Reply:
@emmazzz

Reply:
`new_unsafe_from_bytes` will abort if any of the bytes are not ascii so it's checked already

Reply:
so I'm not sure what's unsafe about this function